lib/active_merchant/billing/gateways/latitude19.rb

```diff
@@ -0,0 +1,411 @@
+        params[:transactionClass] = "eCommerce" || options[:transaction_class]
+      end
+
+      def add_payment_method(params, payment_method)
+        params[:cardExp] = format(payment_method.month, :two_digits).to_s + "/" + format(payment_method.year, :two_digits).to_s
+        params[:cardType] = BRAND_MAP[payment_method.brand.to_s]
+        params[:cvv] = payment_method.verification_value
+        params[:firstName] = payment_method.first_name
+        params[:lastName] = payment_method.last_name
+      end
+
+      def add_customer_data(params, options)
+        if (billing_address = options[:billing_address] || options[:address])
+          params[:address1] = billing_address[:address1]
+          params[:address2] = billing_address[:address2]
+          params[:city] = billing_address[:city]
+          params[:stateProvince] = billing_address[:state]
+          params[:zipPostalCode] = billing_address[:zip]
+          params[:countryCode] = billing_address[:country]
+        end
+      end
+
+      def commit(endpoint, post)
+        begin
+          raw_response = ssl_post(url() + endpoint, post_data(post), headers)
+          response = parse(raw_response)
+        rescue ResponseError => e
+          raw_response = e.response.body
+          response_error(raw_response)
+        rescue JSON::ParserError
+          unparsable_response(raw_response)
+        else
+          success = success_from(response)
+          Response.new(
+            success,
+            message_from(response),
+            response,
+            authorization: success ? authorization_from(response, post[:method]) : nil,
+            avs_result: success ? avs_from(response) : nil,
+            cvv_result: success ? cvv_from(response) : nil,
+            error_code: success ? nil : error_from(response),
+            test: test?
+          )
+        end
+      end
+
+      def headers
+        {
+          "Content-Type"  => "application/json"
+        }
+      end
+
+      def post_data(params)
+        params.to_json
+      end
+
+      def url
+        test? ? test_url : live_url
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def success_from(response)
+        return false if response["result"].nil? || response["error"]
+
+        if response["result"].key?("pgwResponseCode")
+          response["error"].nil? && response["result"]["lastActionSucceeded"] == 1 && response["result"]["pgwResponseCode"] == "100"
+        else
+          response["error"].nil? && response["result"]["lastActionSucceeded"] == 1
+        end
+      end
+
+      def message_from(response)
+        return response["error"] || "HTTP Response Error" unless response.key?("result")
+
+        return response["error"] if response["error"]
+
+        if response["result"].key?("pgwResponseCode")
+          (RESPONSE_CODE_MAPPING[response["result"]["pgwResponseCode"]] || "") + "|" + (response["result"]["responseText"] || "")
```

Comment:
modified to return a single message. Either decoded `pgwResponseCode` or `responseText`